stout/event-loop.cc

```diff
@@ -79,20 +79,34 @@ void EventLoop::Clock::Advance(const std::chrono::nanoseconds& nanoseconds) {
 
 ////////////////////////////////////////////////////////////////////////
 
-static EventLoop* loop = new EventLoop();
+static int8_t loop_memory[sizeof(EventLoop)] = {};
+static EventLoop* loop = reinterpret_cast<EventLoop*>(loop_memory);
+static bool loop_created = false;
 
 ////////////////////////////////////////////////////////////////////////
 
 EventLoop& EventLoop::Default() {
+  if (!loop_created) {
+    new (loop_memory) EventLoop;
+    loop_created = true;
+  }
+
   return *CHECK_NOTNULL(loop);
 }
 
 ////////////////////////////////////////////////////////////////////////
 
-EventLoop& EventLoop::Default(EventLoop* replacement) {
-  delete CHECK_NOTNULL(loop);
-  loop = CHECK_NOTNULL(replacement);
-  return Default();
+void EventLoop::DefaultReset() {
+  if (loop_created) {
+    loop->~EventLoop();
+    for (int i = 0; i < sizeof(EventLoop); i++) {
+      loop_memory[i] = 0;
+    }
```

Comment:
As mentioned by Gorm on the call, [`memset`](https://www.cplusplus.com/reference/cstring/memset/) is probably the right tool to zero out memory.

However, do we need to zero out this memory? Won't the constructor for the new `EventLoop` overwrite it anyway?

Reply:
`new EventLoop` - will not overwrite it.
`new EventLoop()` - will.

Changed the code to use the second one and removed that `for` loop.